app/routes.py

```diff
@@ -1,2 +1,286 @@
+
+
+@customers_bp.route("/<customer_id>", methods=["PUT"], strict_slashes=False)
+def update_customer(customer_id):
+    request_body = request.get_json()
+    customer = Customer.query.get(customer_id)
+    
+    if customer == None:
+        return Response("This customer is not in our database!", status=404)
+
+    elif ("name" not in request_body or "postal_code" not in request_body \
+    or "phone" not in request_body):
+        
+        return jsonify({"details":"Invalid data"}),400
+    
+    elif customer: 
+        form_data = request.get_json()
+
+        customer.name = form_data["name"]
+        customer.postal_code = form_data["postal_code"]
+        customer.phone = form_data["phone"]
+        customer.registered_at = form_data["registered_at"]
+
+        db.session.commit()
+        return jsonify({"customer": customer.customer_to_json()}), 200
+
+
+@customers_bp.route("/<customer_id>", methods=["DELETE"], strict_slashes=False)    
+def delete_single_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+
+    if customer:
+        db.session.delete(customer)
+        db.session.commit()
+        return {
+            "details": f"customer {customer.customer_id} \"{customer.name}\" successfully deleted"}, 200
+    
+    elif customer == None:
+        return Response("This customer is not in our database!", status=404)
+
+
+
+####### VIDEO CRUD #########
+
+@videos_bp.route("", methods=["GET"], strict_slashes=False)
+def get_videos():
+
+#WAVE 3    
+    sort_by_name = request.args.get("sort")
+    if sort_by_name == "asc":
+            videos = Video.query.order_by(Video.name.asc()).all()
+    elif sort_by_name == "desc":
+            videos = Video.query.order_by(Video.name.desc()).all()
+    else:
+        videos = Video.query.all()
+    
+    videos_response = []
+    for video in videos:
+        videos_response.append(video.video_to_json())
+
+    return jsonify(videos_response), 200
+
+
+@videos_bp.route("/<video_id>", methods=["GET"], strict_slashes=False)
+def get_single_video(video_id):
+
+    if not is_int(video_id):
+        return {
+            "message": f"ID {video_id} must be an integer",
+            "success": False
+        }, 400
+
+    video = Video.query.get(video_id)
+
+    if video is not None:
+        return jsonify({"video": video.video_to_json()}), 200
+    else:
+        return Response("This video is not in our database!",status=404)
+
+
+@videos_bp.route("", methods=["POST"], strict_slashes=False)
+def create_video():
+    request_body = request.get_json()
+
+    if ("title" not in request_body or "release_date" not in request_body \
+        or "total_inventory" not in request_body):
+        return jsonify({"details":"Invalid data"}),400
+    
+    else:
+        new_video = Video(title = request_body ["title"],
+                        release_date = request_body["release_date"],
+                        total_inventory = request_body["total_inventory"])
```

Comment:
The current default setting is not a valid option.  The simplest solution is to set available_inventory when creating the video.

```suggestion
        new_video = Video(title = request_body ["title"],
                        release_date = request_body["release_date"],
                        total_inventory = request_body["total_inventory"],
                        available_inventory = request_body["total_inventory"])
```